### PR TITLE
dsv4(npu): support prefill context parallelism with interleave and zigzag

### PR DESCRIPTION
**Pull request by vstone-w**
## Motivation

1、support prefill cp for deepseekv4 on npu platform

## Modifications

Based on the latest community CP code, adapt PCP for DeepSeek-V4:
1. Enable the CP switch on the NPU platform.
2. Perform metadata management inside the DeepSeek-V4 model files, managing two sets—global and local rank—to adapt to the inputs of different operators.

## Accuracy Tests

image

## Speed Tests and Profiling

no cp，tp=16，dp=8, attention_tp=2
cp2，tp=16，dp=8, cp=2

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :hourglass_flowing_sand: [Run #35198971785](https://github.com/sgl-project/sglang/actions/runs/35198971785)
Latest PR Test (Extra): :x: [Run #35198971453](https://github.com/sgl-project/sglang/actions/runs/35198971453)
Latest PR Test (AMD ROCm 10): :hourglass_flowing_sand: [Run #35198971690](https://github.com/sgl-project/sglang/actions/runs/35198971690)

### CODE REVIEW

**Comment by vstone-w** [comment]
image

current commit:
image


all gpu test passed~
i need update commit info. An incorrect rebase caused the commit information to include someone else’s information.

commit info has been corrected，without any file chaning
image

File: python/sglang/srt/models/deepseek_v4.py [hunk 1]
```diff
@@ -3449,7 +3491,6 @@ def determine_num_fused_shared_experts(self):
             0 if is_shared_experts_fusion_disabled() else self.config.n_shared_experts
         )
 
-    @torch.no_grad()
     def forward(
```
Comment (ClownBin) [inline python/sglang/srt/models/deepseek_v4.py:3494]: [P1] Keep the public inference entrypoint under `torch.no_grad()`. The decorator newly added to `self.model.forward` does not cover `self.logits_processor(...)`/LM head. Normal eager decode and non-CP extend call this outer method without an enclosing inference/no-grad context, so the LM head starts recording an autograd graph for each request (and retains its input hidden states). This is a substantial memory/performance regression.

**Review by ClownBin — APPROVED** [review]

File: python/sglang/srt/models/deepseek_v4.py (hunk 1 above)
Comment (vstone-w) [reply python/sglang/srt/models/deepseek_v4.py:3494]: its the common path for eagle decode and cp prefill
